test/test_jit.py

```diff
@@ -4943,6 +4943,23 @@ def named_var_and(x, y):
                 if y is not None and x_none:
                     print(x + y)  # noqa: T484
 
+    def test_optional_tensor(self):
+        @torch.jit.script
+        def fn(x):
+            # type: (Optional[Tensor]) -> int
+            if x is None:
+                return 1
+            else:
+                return 0
+
+            fn(None)
+            g = fn.graph_for(None)
```

Comment:
Add a check that the output of the function is 1 / 0 ?

Reply:
Well, I only added a return value to not have it tell me that it needs to do nothing... :)

Reply:
ya more just as a sanity check that the optimizations i'm adding in this PR don't break this, although i'll wait for the landing of that PR with this one to be safe

Reply:
What I would suggest is that we amend the check to see that the graph is only a constant return for each case. :)